File: LeetCode/Container-With-Most-Water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int maxArea = 0, currentArea = 0;
        int l = 0, r = height.Length - 1;
        while(l<r)
        {
            currentArea = Math.Min(height[l], height[r]) * (r - l);
            maxArea = Math.Max(currentArea, maxArea);
            if (height[l] > height[r])
                r--;
            else
                l++;
        }
        return maxArea;
    }
}

File: LeetCode/Container-With-Most-Water.cpp
1class Solution {
2public:
3    int maxArea(vector<int>& height) {
4        int l=0, r = height.size() -1;
5        int area, maxArea = 0;
6        while(l < r){
7            area = min(height[l], height[r]) * (r - l);
8            maxArea = max(maxArea, area);
9
10            if(height[l]<height[r]) l++; else r--;
11        }
12        return  maxArea;
13    }
14};